plugins/woocommerce/templates/single-product/add-to-cart/grouped.php

```diff
@@ -64,7 +64,17 @@
 								woocommerce_template_loop_add_to_cart();
 							} elseif ( $grouped_product_child->is_sold_individually() ) {
 								echo '<input type="checkbox" name="' . esc_attr( 'quantity[' . $grouped_product_child->get_id() . ']' ) . '" value="1" class="wc-grouped-product-add-to-cart-checkbox" id="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" />';
-								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' . esc_html__( 'Buy one of this item', 'woocommerce' ) . '</label>';
+								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' .
+									sprintf(
+										/* translators: %1$s: Product name. %2$s: Product price. */
+										$grouped_product_child->is_on_sale() ?
+											esc_html__( 'Buy one of %1$s on sale for %2$s, original price was %3$s', 'woocommerce' ) :
+											esc_html__( 'Buy one of %1$s for %2$s', 'woocommerce' ),
+										esc_html( $grouped_product_child->get_name() ),
+										esc_html( wp_strip_all_tags( wc_price( $grouped_product_child->get_price() ) ) ),
+										$grouped_product_child->is_on_sale() ? esc_html( wp_strip_all_tags( wc_price( $grouped_product_child->get_regular_price() ) ) ) : ''
+									) .
+								'</label>';
```

Comment:
Do you think it would make sense to separate the translations so that translators can be more precise? For example:

```suggestion
								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">';
								if ( $grouped_product_child->is_on_sale() ) {
									printf(
										/* translators: %1$s: Product name. %2$s: Sale price. %3$s: Regular price */
										esc_html__( 'Buy one of %1$s on sale for %2$s, original price was %3$s', 'woocommerce' ),
										esc_html( $grouped_product_child->get_name() ),
										esc_html( wp_strip_all_tags( wc_price( $grouped_product_child->get_price() ) ) ),
										esc_html( wp_strip_all_tags( wc_price( $grouped_product_child->get_regular_price() ) ) )
									);
								} else {
									printf(
										/* translators: %1$s: Product name. %2$s: Product price */
										esc_html__( 'Buy one of %1$s for %2$s', 'woocommerce' ),
										esc_html( $grouped_product_child->get_name() ),
										esc_html( wp_strip_all_tags( wc_price( $grouped_product_child->get_price() ) ) )
									);
								}
								echo '</label>';

```

Reply:
@WunderBart yesss, I think that's better! Thank you 🙇